src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/ArrayBufferWriter.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.Text.Json
+{
+    internal sealed class ArrayBufferWriter<T> : IBufferWriter<T>
+    {
+        private T[] _buffer;
+        private int _index;
+
+        private const int MinimumBufferSize = 256;
+
+        public ArrayBufferWriter()
+        {
+            _buffer = new T[MinimumBufferSize];
+            _index = 0;
+        }
+
+        public ArrayBufferWriter(int initialCapacity)
```

Comment:
Does `initialCapacity` need to be signed?

Reply:
It doesn't need to be, but we generally have capacity like this be signed int32:
https://github.com/dotnet/corefx/issues/34894

https://github.com/dotnet/corefx/blob/1fd3717d73da3ae7440841e8fded4adbba5b0770/src/System.Collections.NonGeneric/src/System/Collections/Stack.cs#L42

Reply:
We do that in public types because the unsigned types aren't CLS complaint. That said, I don't think it matters enough here to bother changing.

Reply:
Fair enough?